cluster/cluster.go

```diff
@@ -226,6 +226,50 @@ func LogFatalClusterError(errMessage string, scope int, numErrors int) {
 	gplog.Fatal(errors.Errorf("%s on %d %s. See %s for a complete list of errors.", errMessage, numErrors, segMsg, gplog.GetLogFilePath()), "")
 }
 
+/*
+ * ExecuteMasterClusterCommands is a generic wrapper function
+ * to simplify execution of local commands pertaining to segments.
+ * e.g. running scp locally many times to push files to all segments
+ */
+func (cluster *Cluster) ExecuteMasterClusterCommands(generateCommand func(int) string) *RemoteOutput {
```

Comment:
I do like the argument of the anonymous function, but please weigh that with reuse benefits: the similarity between this method and the existing ExecuteClusterCommand(): is there an opportunity for reuse?

Reply:
+1 to this. The logic can be combined. I think i'll still take an anonymous function though.